runtime/lua/vscode/sync-options.lua

```diff
@@ -46,68 +65,67 @@ end
-  end
-  if vim.bo[buf].et ~= opts.insertSpaces then
-    vim.bo[buf].et = opts.insertSpaces
-  end
 
-  local win
-  for _, w in ipairs(api.nvim_list_wins()) do
-    local win_buf = api.nvim_win_get_buf(w)
-    if win_buf == buf then
-      win = w
-      break
-    end
-  end
+  M.buf_set_option(buf, "tabstop", opts.tabSize)
+  M.buf_set_option(buf, "shiftwidth", opts.tabSize)
+  M.buf_set_option(buf, "expandtab", opts.insertSpaces)
 
-  if win then
+  local win = api.nvim_get_current_win()
+  if api.nvim_win_get_buf(win) == buf then
     set_number(win, opts.lineNumbers)
   end
 end
 
 ---Check changes from nvim
 ---Set vscode options
-local function _check_options()
-  ---@type EditorOptions?
-  local opts = vim.b.vscode_editor_options
-  if not opts then -- should not happen
+local function check_options()
+  local vscode_opts = vim.b.vscode_editor_options
+  if not vscode_opts then -- should not happen
     return
   end
-  if not vim.b.vscode_editor_options_first_checked then -- load the defaults
+
+  if not vim.b.vscode_editor_options_first_checked then --load the defaults
     vim.b.vscode_editor_options_first_checked = true
-    vim.bo.ts = opts.tabSize
-    vim.bo.sw = opts.tabSize
-    vim.bo.expandtab = opts.insertSpaces
-    set_number(0, opts.lineNumbers)
+    M.buf_set_option(0, "tabstop", vscode_opts.tabSize)
+    M.buf_set_option(0, "shiftwidth", vscode_opts.tabSize)
+    M.buf_set_option(0, "expandtab", vscode_opts.insertSpaces)
+    set_number(0, vscode_opts.lineNumbers)
     return
   end
 
-  local ts, sw, et = vim.bo.ts, vim.bo.sw, vim.bo.et
-  local lineNumbers = get_number_style(vim.wo.nu, vim.wo.rnu)
+  local buf = api.nvim_get_current_buf()
+  local win = api.nvim_get_current_win()
+  local ts = api.nvim_get_option_value("tabstop", { buf = buf })
+  local sw = api.nvim_get_option_value("shiftwidth", { buf = buf })
+  local et = api.nvim_get_option_value("expandtab", { buf = buf })
 
   if sw ~= ts then
-    vim.bo.sw = ts -- must be the same
+    M.buf_set_option(buf, "shiftwidth", ts) -- must be the same
   end
 
-  if ts ~= opts.tabSize or et ~= opts.insertSpaces or lineNumbers ~= opts.lineNumbers then
-    opts.tabSize = ts
-    opts.insertSpaces = et
-    opts.lineNumbers = lineNumbers
-    vim.b.vscode_editor_options = opts
-    vscode.action("set_editor_options", { args = { api.nvim_get_current_buf(), opts } })
+  local nvim_opts = {
+    tabSize = ts,
+    insertSpaces = et,
+    lineNumbers = get_number_style(
+      api.nvim_get_option_value("number", { win = win }),
+      api.nvim_get_option_value("relativenumber", { win = win })
+    ),
+  }
+
+  if vim.deep_equal(nvim_opts, vscode_opts) then
+    return
   end
+
+  vim.b.vscode_editor_options = nvim_opts
+  vscode.action("set_editor_options", { args = { api.nvim_get_current_buf(), nvim_opts } })
 end
 
-local check_options = util.debounce(_check_options, 20)
+local check_options_debounced = util.debounce(check_options, 20)
```

Comment:
could avoid some churn by not renaming this. the old name seemed fine

Reply:
Fixed